test/e2e/suite/issuers/acme/certificate/http01.go

```diff
@@ -222,4 +226,49 @@ var _ = framework.CertManagerDescribe("ACME Certificate (HTTP01)", func() {
 		err = h.WaitCertificateIssuedValid(f.Namespace.Name, certificateName, time.Minute*5)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	It("should obtain a signed certificate with a single CN from the ACME server after solver pod killed", func() {
+		certClient := f.CertManagerClientSet.CertmanagerV1alpha1().Certificates(f.Namespace.Name)
+
+		By("Creating a Certificate")
+		_, err := certClient.Create(util.NewCertManagerACMECertificate(certificateName, certificateSecretName, issuerName, v1alpha1.IssuerKind, nil, nil, acmeIngressClass, acmeIngressDomain))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("killing the solver pod")
+		podClient := f.KubeClientSet.CoreV1().Pods(f.Namespace.Name)
+		var pod corev1.Pod
+		err = wait.PollImmediate(500*time.Millisecond, time.Minute,
+			func() (bool, error) {
+				log.Logf("Waiting for solver pod to exist")
+				podlist, err := podClient.List(metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+
+				for _, p := range podlist.Items {
+					log.Logf("solver pod %s", p.Name)
+					// TODO(dmo): make this cleaner instead of just going by name
+					if strings.Contains(p.Name, "http-solver") {
```

Comment:
:+1: not important for this PR really, but we can use a label selector in the List call above to ensure we get the correct pods :smile: (similar to how we do it in ensurePod)